src/binder/expression_binder.cpp

```diff
@@ -79,6 +81,23 @@ std::shared_ptr<Expression> ExpressionBinder::implicitCastIfNecessary(
     return implicitCast(expression, DataType(targetTypeID));
 }
 
+std::shared_ptr<Expression> ExpressionBinder::implicitCast(
+    const std::shared_ptr<Expression>& expression, const common::DataType& targetType) {
+    if (expression->dataType.typeID != VAR_LIST && expression->dataType.typeID != FIXED_LIST &&
```

Comment:
Ideally when you want your getCastCost function to take data type instead of data type ID, so that it can also return MAX cost for FIXED_LIST/VAR_LIST

Reply:
added a new getCastCost function